cmake/Modules/VolkPython.cmake

```diff
@@ -36,12 +36,16 @@ if(PYTHON_EXECUTABLE)
 else(PYTHON_EXECUTABLE)
 
     #use the built-in find script
-    set(Python_ADDITIONAL_VERSIONS 3.4 3.5 3.6)
-    find_package(PythonInterp 2)
+    set(Python_ADDITIONAL_VERSIONS 3.4 3.5 3.6 3.7 3.8)
+    find_package(PythonInterp 3)
```

Comment:
according to CMake 3.16.0's `FindPythonInterp.cmake`:
```
deprecated:: 3.12
  Use :module:`FindPython3`, :module:`FindPython2` or :module:`FindPython` instead.
```
Looks like the new usage is `find_package(Python3 COMPONENTS Interpreter)`

Same for the 2.7 call.

Reply:
Looks like instead of `` we now do `set(Python3_FIND_STRATEGY VERSION)` ... which is the default so we  can just remove the `Python_ADDITIONAL_VERSIONS` I think.
```
``Python3_FIND_STRATEGY``
  This variable defines how lookup will be done.
  The ``Python3_FIND_STRATEGY`` variable can be set to empty or one of the
  following:

  * ``VERSION``: Try to find the most recent version in all specified
    locations.
    This is the default if policy :policy:`CMP0094` is undefined or set to
    ``OLD``.
  * ``LOCATION``: Stops lookup as soon as a version satisfying version
    constraints is founded.
    This is the default if policy :policy:`CMP0094` is set to ``NEW``.
```

Reply:
Yeah, the new, in fact too new, way would be `find_package(Python COMPONENTS Interpreter)` [cf here](https://cmake.org/cmake/help/v3.16/module/FindPython.html#module:FindPython). This call would prefer Py3k over Py2k, thus effectively implement the same behavior I intend to implement.
I browsed through the CMake documentation and it seems this function was only added in CMake 3.12. And we currently require CMake 3.8. 

With CMake 3.12+ we could simply do this:
```
if(PYTHON_EXECUTABLE)
    set(Python_EXECUTABLE ${PYTHON_EXECUTABLE})
else(PYTHON_EXECUTABLE)
    find_package(Python COMPONENTS Interpreter)
endif(PYTHON_EXECUTABLE)

#make the path to the executable appear in the cmake gui
set(PYTHON_EXECUTABLE ${Python_EXECUTABLE} CACHE FILEPATH "python interpreter")
```
Since `grep -irnI PYTHONINTERP_FOUND .` did not reveal any use of this variable outside this section, we don't need to set this variable anymore.

I suggest to stick with my implementation for now. We can update this section after we move to CMake 3.12+ as a minimum. I don't like to make this section version dependent. This would make it more difficult to read.
@michaelld how would you proceed?

Reply:
OK yeah given the CMake min version, let's leave this code as it is. Would you please add a note about changing up the code when CMake >= 3.12? That way we're leaving something useful behind!